### PR TITLE
merged all commits into one

### PR DESCRIPTION
**Pull request by MysticDakra**

### CODE REVIEW

File: test/seriesController/seriesScenarios.ts [hunk 1]
```diff
@@ -53,10 +53,8 @@ contract("Series Scenarios", (accounts) => {
     // Give Alice the collateral
     await collateralToken.mint(aliceAccount, mintAmount)
 
-    // Erase all previous rounds
-    await deployedMockPriceOracle.reset()
     const oraclePrice = 22_000 * 10 ** 8 // 22k
-    await deployedMockPriceOracle.addRound(oraclePrice, expiration, expiration)
+    await deployedMockPriceOracle.setLatestAnswer(oraclePrice)
```
Comment (lukaskiss222) [inline test/seriesController/seriesScenarios.ts:57]: @MysticDakra have you done this change? I can not remember, if it was done by myself.

File: contracts/amm/AmmDataProvider.sol [hunk 2]
```diff
@@ -608,14 +657,16 @@ contract AmmDataProvider is IAmmDataProvider {
...
             optionTokenGetCollateralOut(
                 seriesId,
                 ammAddress,
                 wTokenAmount,
                 amm.collateralBalance(),
-                getPriceForSeries(seriesId, amm.getVolatility(seriesId)),
+                getPriceForSeries(
+                    seriesController.series(seriesId),
+                    amm.getVolatility(seriesId)
+                ),
```
Comment (lukaskiss222) [inline contracts/amm/AmmDataProvider.sol:669]: line 667. Is it ok to pass it directly?

File: contracts/series/SeriesController.sol [hunk 3]
```diff
@@ -177,19 +177,19 @@ contract SeriesController is
             if (settlementPrice >= currentSeries.strikePrice) {
                 // OTM
                 writerShare = getCollateralPerOptionToken(
-                    _seriesId,
+                    currentSeries,
```
Comment (lukaskiss222) [inline contracts/series/SeriesController.sol:180]: SeriesController.sol and  ISeriesController.sol  should use the same naming as AmmDataProvider.sol for parameters: currentSeries -> series

File: contracts/series/SeriesController.sol [hunk 4]
```diff
@@ -411,36 +411,34 @@ contract SeriesController is
     }
 
     /// @notice Given a series ID and an amount of bToken/wToken, return the amount of collateral token received when it's exercised
-    /// @param _seriesId The Series ID
+    /// @param _currentSeries The Series
     /// @param _optionTokenAmount The amount of bToken/wToken
     /// @return The amount of collateral token received when exercising this amount of option token
     function getCollateralPerOptionToken(
-        uint64 _seriesId,
+        ISeriesController.Series memory _currentSeries,
```
Comment (lukaskiss222) [inline contracts/series/SeriesController.sol:418]: the same

File: contracts/series/SeriesController.sol [hunk 5]
```diff
@@ -411,36 +411,34 @@ contract SeriesController is
...
     /// @dev Given a Series and an amount of underlying, return the amount of collateral adjusted for decimals
     /// @dev In almost every callsite of this function the price is equal to the strike price, except in Series.getSettlementAmounts where we use the settlementPrice
-    /// @param _seriesId The Series ID
+    /// @param _currentSeries The Series
     /// @param _underlyingAmount The amount of underlying
     /// @param _price The price of the collateral token in units of price token
     /// @return The amount of collateral
     function getCollateralPerUnderlying(
-        uint64 _seriesId,
+        ISeriesController.Series memory _currentSeries,
```
Comment (lukaskiss222) [inline contracts/series/SeriesController.sol:436]: Same

File: test/seriesController/seriesScenarios.ts (hunk 1 above)
Comment (MysticDakra) [reply test/seriesController/seriesScenarios.ts:57]: I didnt make any changes I just pulled these in from your other branch.

File: contracts/amm/AmmDataProvider.sol (hunk 2 above)
Comment (MysticDakra) [reply contracts/amm/AmmDataProvider.sol:669]: I believe this is okay @seafi do you have any issues with this here?

File: contracts/series/SeriesController.sol (hunk 3 above)
Comment (MysticDakra) [reply contracts/series/SeriesController.sol:180]: yes lets make these both series

**Comment by MysticDakra** [comment]
These changes look good to me @lukaskiss222 lets make one more pass through looking at gas optimizations and any other issues before we finalize this. 
Other things to look for is rentrancy attacks, or other hacks.

File: test/seriesController/seriesScenarios.ts (hunk 1 above)
Comment (lukaskiss222) [reply test/seriesController/seriesScenarios.ts:57]: @MysticDakra, I realised what happened. 
The removed code was added to the main in commit 7b97bba095d8706fd0e67e2fa5f5167af02c2637.
Unfortunately, this branch was forked before that commit and does not contain the new changes after this commit. (It was not synced with main)
So it overrides the new changes with the old ones.

**Review by lukaskiss222 — APPROVED** [review]

**Review by seafi — APPROVED** [review]

**Pull request merged**